INTEGRATING.md

```diff
@@ -1,36 +1,171 @@
+```
+
+Grafana Agent may forward client charms metrics and associated alert rules (that are received using the `metrics-endpoint` relation) to Prometheus using the [`prometheus_remote_write`](https://charmhub.io/prometheus-k8s/libraries/prometheus_remote_write) interface.
+To do that you need to relate Grafana Agent with Prometheus:
+
+```bash
+juju relate grafana-agent-k8s:send-remote-write prometheus-k8s
+```
+
+
+Once these relations are established, you can check that by running:
+
+```bash
+juju status --relations
+
+Model  Controller  Cloud/Region        Version  SLA          Timestamp
+docs   charm-dev   microk8s/localhost  2.9.32   unsupported  17:55:18-03:00
+
+App                Version  Status   Scale  Charm              Channel    Rev  Address         Exposed  Message
+grafana                     active       1  grafana-k8s        edge        40  10.152.183.168  no
+grafana-agent-k8s           active       1  grafana-agent-k8s  edge        11  10.152.183.123  no
+loki-k8s                    waiting      1  loki-k8s           edge        41  10.152.183.43   no
+prometheus-k8s     2.33.5   active       1  prometheus-k8s     candidate   55  10.152.183.128  no
+zinc-k8s           0.2.8    active       1  zinc-k8s           stable      26  10.152.183.247  no
+
+Unit                  Workload  Agent       Address       Ports  Message
+grafana-agent-k8s/0*  active    idle        10.1.157.120
+grafana/0*            active    idle        10.1.157.88
+loki-k8s/0            active    idle        10.1.157.86
+prometheus-k8s/0*     active    idle        10.1.157.89
+zinc-k8s/0*           active    idle        10.1.157.79
+
+Relation provider                        Requirer                             Interface                Type     Message
+grafana-agent-k8s:grafana-dashboard      grafana:grafana-dashboard            grafana_dashboard        regular
+grafana-agent-k8s:logging-provider       zinc-k8s:logging                     loki_push_api            regular
+grafana-agent-k8s:self-metrics-endpoint  prometheus-k8s:metrics-endpoint      prometheus_scrape        regular
+grafana:grafana                          grafana:grafana                      grafana_peers            peer
+loki-k8s:logging                         grafana-agent-k8s:logging-consumer   loki_push_api            regular
+prometheus-k8s:prometheus-peers          prometheus-k8s:prometheus-peers      prometheus_peers         peer
+prometheus-k8s:receive-remote-write      grafana-agent-k8s:send-remote-write  prometheus_remote_write  regular
+zinc-k8s:metrics-endpoint                grafana-agent-k8s:metrics-endpoint   prometheus_scrape        regular
+```
 
 
 ## Deployment scenarios
+
 ```mermaid
 flowchart LR
     subgraph COS Lite
         prometheus
         loki
+        grafana
     end
     subgraph K8s cluster
         prometheus               ---      |remote_write| grafana-agent
+        prometheus               ---      |prometheus_scrape| grafana-agent
```

Comment:
Iiuc, this line shouldn't be here:
- Both prom and agent have `metrics-endpoint` under requires
- The agent is "converting" scrape targets into remote write

Reply:
In fact this `prmetheus_scrape` is the interface used by `self-metrics-endpoint` relation, with this Prometheus can scrape Grafana Agent self metrics.

Should we show relation names too?

Reply:
Maybe we should not show self-metrics relations...

Reply:
Yeah I think in the context of this deployment scenarios it is unnecessary.
I would certainly add the self monitoring relations in the bundle doc though.

Other kinds of self monitoring though could be worth mentioning here, but I think it would be a separate diagram. For example the resilience @rbarry82 was mentioning a while ago.